### PR TITLE
Refs #32502 -- Avoided table rebuild when adding fields with no default on SQLite.

### PR DESCRIPTION
**Pull request by felixxm**
This covers using `ALTER TABLE ADD COLUMN` statement.

ticket-32502

### CODE REVIEW

**Review by charettes — APPROVED** [review]
Looking good, what the origin of the initial spatialite failures?

**Comment by felixxm** [comment]
Thanks for checking :+1: 

> Looking good, what was the origin of the initial spatialite failures?

`sql_create_column_inline_fk` was missing.

**Review by ngnpope — APPROVED** [review]

File: tests/schema/tests.py [hunk 1]
```diff
@@ -517,6 +517,15 @@ def test_add_field(self):
...
         self.assertFalse(any(drop_default_sql in query['sql'] for query in ctx.captured_queries))
+        # Table is not rebuilt.
+        self.assertIs(any(
+            'CREATE TABLE' in query['sql']
+            for query in ctx.captured_queries
+        ), False)
+        self.assertIs(any(
+            'DROP TABLE' in query['sql']
+            for query in ctx.captured_queries
+        ), False)
```
Comment (adamchainz) [inline tests/schema/tests.py:528]: why not `assertFalse` ?

```suggestion
        self.assertFalse(any(
            'CREATE TABLE' in query['sql']
            for query in ctx.captured_queries
        )
        self.assertFalse(any(
            'DROP TABLE' in query['sql']
            for query in ctx.captured_queries
        ))
```

Comment (ngnpope) [reply tests/schema/tests.py:528]: `assertFalse()` matches anything falsy. If you want to check for `False` only you need to use `assertIs(…, False)`.

Comment (adamchainz) [reply tests/schema/tests.py:528]: But `any()` always returns a bool? https://docs.python.org/3/library/functions.html#any

**Pull request merged**